### PR TITLE
Community and Event card components #9

### LINKED ISSUE #9: Parametrize and abstract Card.astro component
We are deploying the almost same card component inside Community and Events tab. We should abstract both components and unify them in a Card.astro component.

I was thinking we should provide some optional props for the Astro component. For example, describa a `? long_desc`, `? url`, and the like.

I do know https://github.com/onwidget/astrowind have some components that have the mentioned behaviour.

### PR DESCRIPTION
**Pull request by nsandoya**
The original request was merge in one card the components for communities and events pages. While working in it, I realized we actually need two separated components: each one requires a diferent interface for its props.

Based in the existing code, I created communty and event card components, made interfaces for both, and implemented the new components in _communities_ and _events_ pages

Resolves #9

### CODE REVIEW

**Comment by bjgavilanes** [comment]
I'm not that sure if we would want to split, or maintain, two implementations for community and event each.

I envisioned to codify an astro component that grabs a JSON -- let's say, community.json or events.json -- and deploys the cards components with provided information. Sure, requirements for community or events as JSON files do differ; but!, we would type for optional keys at the astro component, and make the same component work for both use cases.

However, as I review this PR, I do concede that events/community distinction does provide a level of desired granularity... But, I have to insist that we can accomplish if we supported optional props in the astro component.

@IvanM9 @davcortez got some thoughs?

**Comment by nsandoya** [comment]
Hi Brayen,

I understand your vision about this component. Maybe we can merge both points of view: What do you think about creating a _generic component_ and two _derived components_ by it? :

1. **CardBase** (_generic component_): We can define the common card elements in it --as I understand: banner image, main title, and description. Then, we can include an `<slot/>` in this base card.
2. **CardEvent and CardCommunity: ** We can use CardBase to create _derivated components_ that can manage contextual differences through their own interfaces and props --and include more specific style changes over time if needed.

This proposal aims to avoid overload in code while preserving the parametric approach envisioned :)

**Comment by bjgavilanes** [comment]
Hi, Nataly!, how you doing?, I'm hella sleepy rn, send home to sleep.

On Mon Dec 9, 2024 at 11:25 AM -03, Nataly Sandoya wrote:
> Hi Brayen,
>
> I understand your vision about this component. Maybe we can merge both points of view: What do you think about creating a _generic component_ and two _derived components_ by it? :
>
> 1. **CardBase** (_generic component_): We can define the common card elements in it --as I understand: banner image, main title, and description. Then, we can include an `<slot/>` in this base card.
> 2. **CardEvent and CardCommunity**: We can use CardBase to create _derivated components_ that can manage contextual differences through their own interfaces and props --and include more specific style changes over time if needed.
>
> This proposal aims to avoid overload in code while preserving the parametric approach envisioned :)


I actually like this approach. Thank you for everything.

File: src/interfaces/events.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export interface Event {
+    title: string,
+    description: string,
+    links: string[],
```
Comment (IvanM9) [inline src/interfaces/events.ts:4]: I recommend having only one link to redirect to the selected event page, and then in the event-name.json file, include all the links you want to display.

```suggestion
    link: string,
```

**Review by IvanM9 — CHANGES_REQUESTED** [review]

File: src/interfaces/events.ts (hunk 1 above)
Comment (bjgavilanes) [reply src/interfaces/events.ts:4]: At `src/interfaces/communities` we got a similar behaviour. Would you recommend to implement this change in that source file, too?

Comment (nsandoya) [reply src/interfaces/events.ts]: Thank you, guys 🐥

**Comment by nsandoya** [comment]
Hi Brayen,

The new proposal was implemented 🐥

**Comment by bjgavilanes** [comment]
Deployed and tested locally. Thank you for everything!,
Brayen

P.S. Just a heads up: this repo is intented to have a trunk-based development workflow. That means you should target to main; UNLESS (!!!), you are making an experimental or unstable change -- such as my shell.nix and nix package manager implementation --.

Again, thank you for everything!,
Brayen

**Pull request merged**